include/bl_validation.h

```diff
@@ -74,14 +74,15 @@ struct bl_validate_fw_ext_api {
  */
 int set_monotonic_version(uint16_t version, uint16_t slot);
 
-
 /** Read 15 bit version and 1 bit slot from a 16 bit monotonic counter.
  *
  * @param[out]  slot_out  Slot where firmware is located. Can be NULL.
+ * @param[out]  version   Firmware version. Can be any unsigned 15 bit value.
  *
- * @return Firmware version. Can be any unsigned 15 bit value.
+ * @retval 0       Success
+ * @retval -EINVAL Error during reading the version or *version is NULL.
  */
-uint16_t get_monotonic_version(uint16_t *slot_out);
+int get_monotonic_version(uint16_t *slot_out, uint16_t *version);
```

Comment:
I don't know if it has any value but would it make sense to create a `struct uint15_t` to wrap this in so that the API will be a bit harder to misuse?